src/java.base/share/classes/jdk/internal/classfile/ClassReader.java

```diff
@@ -163,6 +172,96 @@ public sealed interface ClassReader extends ConstantPool
      */
     ClassEntry readClassEntry(int offset);
 
+    /**
+     * {@return the constant value entry whose index is given at the specified
+     * offset within the classfile}
+     * @param offset the offset of the index within the classfile
+     * @throws ConstantPoolException if the index is out of range of the
+     *         constant pool size, or zero, or the index does not correspond to
+     *         a constant value entry
+     */
+    ConstantValueEntry readConstantValueEntry(int offset);
+
```

Comment:
I'm not sure that all this additional API surface carries its weight in ClassReader, when each of these methods just called readEntry() and casts to the desired type?  I can see two alternate ways here:

 - Ordinary casts at the use site (worse error handling, but ultimately the same result), or
 -  A single method 

<T extends ConstantPoolEntry> T readEntry(int pos, Class<T> t)

Reply:
We need at least `<T extends PoolEntry> T readEntry(int pos, Class<T> t)` to throw `ConstantPoolException` instead of `ClassCastException` from broken entries.